--- a/common.h	2025-11-25 09:02:40.636109060 +0900
+++ b/common.h	2025-11-25 09:04:00.397115875 +0900
@@ -11,7 +11,11 @@
 #include <ext/standard/php_var.h>
 #include <ext/standard/php_math.h>
 #include <zend_smart_str.h>
+#if PHP_VERSION_ID >= 80500
+#include "Zend/zend_smart_string.h"
+#else
 #include <ext/standard/php_smart_string.h>
+#endif
 
 #define PHPREDIS_GET_OBJECT(class_entry, o) (class_entry *)((char *)o - XtOffsetOf(class_entry, std))
 #define PHPREDIS_ZVAL_GET_OBJECT(class_entry, z) PHPREDIS_GET_OBJECT(class_entry, Z_OBJ_P(z))
